Sources/tinys3/model/AWSProfileConfig.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+/// A list of key-value pairs representing AWS configuration values for a given (single) profile
+struct AWSProfileConfig {
+    let values: [String: String]
+
+    enum Key: String {
+        case accessKeyId = "aws_access_key_id"
+        case secretKey   = "aws_secret_access_key"
+        case region      = "region"
+    }
+
+    subscript(key: String) -> String? {
+        return self.values[key]
+    }
+
+    subscript(key: Key) -> String? {
+        return self[key.rawValue]
+    }
+}
+
+extension AWSProfileConfig {
+    /// Parses the `~/.aws/credentials` file for profiles and their associated configuration values
+    /// - Returns: A dictionary containing an `AWSProfileConfig` for each profile name found
+    static func profilesFromCredentialsUserFile() throws -> [String: AWSProfileConfig] {
+        let url = FileManager.default.homeDirectoryForCurrentUser
+            .appendingPathComponent(".aws")
+            .appendingPathComponent("credentials")
+        return try profiles(from: url, isCredentialsFile: true)
+    }
+
+    /// Parses the `~/.aws/config` file for profiles and their associated configuration values
+    /// - Returns: A dictionary containing an `AWSProfileConfig` for each profile name found
+    static func profilesFromConfigUserFile() throws -> [String: AWSProfileConfig] {
+        let url = FileManager.default.homeDirectoryForCurrentUser
+            .appendingPathComponent(".aws")
+            .appendingPathComponent("config")
+        return try profiles(from: url, isCredentialsFile: false)
+    }
+
+    /// Parses the profiles and their associated configuration values from a given file
+    /// - Parameters:
+    ///   - path: The URL of the file to parse
+    ///   - isCredentialsFile: Indicates if the file to parse is a credentials file (true) or a config file (false)
+    /// - Returns: A dictionary containing an `AWSProfileConfig` for each profile name found
+    static func profiles(from path: URL, isCredentialsFile: Bool) throws -> [String: AWSProfileConfig] {
```

Comment:
IMHO we should change `isCredentialsFile` to an enum like `fileType` to handle other potential cases (or even if it's only ever the two cases, it's easier to understand what's going on when branching on the file type.

Reply:
Addressed by afca280eeb5ecfadbd0f82ed0ee7de12cd11ee9b